packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -217,11 +217,22 @@ class EditableTextState extends State<EditableText> implements TextInputClient {
   TextSelectionOverlay _selectionOverlay;
 
   final ScrollController _scrollController = new ScrollController();
+  bool _didAutoFocus = false;
 
   @override
   void initState() {
     super.initState();
     _currentValue = config.value;
+    config.focusNode.addListener(_handleFocusChanged);
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!_didAutoFocus && config.autofocus) {
+      FocusScope.of(context).autofocus(config.focusNode);
+      _didAutoFocus = true;
+    }
```

Comment:
The current design leaves you focused in the old scope.  You'll move to the new scope the next time you become focused.  I wasn't sure what behavior we wanted in this case.  Presumably if you stay in the same scope, you should remain focused but if you move to a new scope then maybe you should become unfocused?